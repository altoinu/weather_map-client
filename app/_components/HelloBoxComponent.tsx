"use client";

import { Button, Stack } from "@mui/material";
import { useState } from "react";

type HomePageProps = {
  personName: string;
  secondPersonName?: string;
  /**
   * Callback fired when the button is clicked.
   *
   * @param {Number} buttonNum Button number clicked.
   */
  onButtonClick?: (buttonNum: number) => void;
};

const HelloBoxComponent = ({
  personName,
  onButtonClick,
  ...props
}: HomePageProps) => {
  const [clickedButtonResponse, setClickedButtonResponse] = useState<string>();

  const handleButtonClick = (buttonNum: number) => {
    const response =
      buttonNum === 1
        ? "First button clicked"
        : buttonNum === 2
          ? "Second button clicked"
          : "something else clicked";

    setClickedButtonResponse(response);

    onButtonClick?.(buttonNum);
  };

  return (
    <>
      <h1>
        Hello {personName}
        {props.secondPersonName ? ` and ${props.secondPersonName}` : ""}!
      </h1>
      {onButtonClick && (
        <Stack direction="row" gap={1}>
          <Button variant="contained" onClick={() => handleButtonClick(1)}>
            Click Me!
          </Button>
          <Button variant="contained" onClick={() => handleButtonClick(2)}>
            Click Me Too!
          </Button>
        </Stack>
      )}
      {clickedButtonResponse && (
        <p>clicked button response: {clickedButtonResponse}</p>
      )}
    </>
  );
};

export default HelloBoxComponent;
