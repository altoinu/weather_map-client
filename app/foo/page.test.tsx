import HelloBoxComponent from "../_components/HelloBoxComponent";
import Page from "./page";
import { render, screen } from "@testing-library/react";
import { usePathname } from "next/navigation";

jest.mock("next/navigation");
const mockUsePathname = usePathname as jest.Mock;

//<h2>This is mocked HelloBoxComponent.</h2>
jest.mock("../_components/HelloBoxComponent");

describe("foo page", () => {
  beforeAll(() => {
    mockUsePathname.mockReturnValue("/mockURL");
  });

  it("should render", () => {
    const { container: pageContainer } = render(<Page />);

    expect(pageContainer).toMatchSnapshot();
  });

  it("should render a heading", () => {
    render(<Page />);

    const heading = screen.getByRole("heading");
    expect(heading.textContent).toBe("This is Foo page.");
  });

  it("should render HelloBoxComponents", () => {
    render(<Page />);

    expect(HelloBoxComponent).toHaveBeenCalledTimes(2);
    expect(HelloBoxComponent).toHaveBeenNthCalledWith(
      1,
      expect.objectContaining({ personName: "Kaoru" }),
      undefined,
    );
    expect(HelloBoxComponent).toHaveBeenNthCalledWith(
      2,
      expect.objectContaining({ personName: "John", secondPersonName: "Doe" }),
      undefined,
    );
  });
});
