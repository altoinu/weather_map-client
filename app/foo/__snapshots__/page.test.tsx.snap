// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foo page should render 1`] = `
<div>
  <h2>
    This is Foo page.
  </h2>
  <p>
    Pathname: 
    /mockURL
  </p>
  <br />
  <br />
  <br />
  <a
    href="/mockURL/bar"
    style="color: red; text-decoration: underline;"
  >
    to Bar page
  </a>
</div>
`;
